mvpa2/algorithms/searchlight_hyperalignment_surface.py

```diff
@@ -0,0 +1,564 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Searchlight-based hyperalignment"""
+import os, tempfile
+import numpy as np
+from mvpa2.base.state import ClassWithCollections
+from mvpa2.base.param import Parameter
+from mvpa2.algorithms.hyperalignment import Hyperalignment
+from mvpa2.measures.base import Measure
+from mvpa2.datasets import Dataset
+from mvpa2.mappers.staticprojection import StaticProjectionMapper
+from scipy.linalg import LinAlgError
+from mvpa2.mappers.zscore import zscore
+from scipy.sparse import coo_matrix, csc_matrix
+#from mvpa2.featsel.helpers import FractionTailSelector
+from mvpa2.featsel.helpers import FixedNElementTailSelector
+import mvpa2.misc.surfing.volgeom as volgeom
+from mvpa2.support.nibabel import surf
+import mvpa2.misc.surfing.volume_mask_dict as volume_mask_dict
+import mvpa2.misc.surfing.surf_voxel_selection as surf_voxel_selection
+from mvpa2.misc.surfing.queryengine import SurfaceVerticesQueryEngine
+from mvpa2.algorithms.searchlight_hyperalignment import compute_feature_scores
+from mvpa2.base.progress import ProgressBar
+from mvpa2.support import copy
+from mvpa2.base.hdf5 import h5save, h5load
+import mvpa2
+if __debug__:
+    from mvpa2.base import debug
+
+class HyperalignmentMeasure(Measure):
```

Comment:
oh -- djavu!  Yet another one!  how is this one different? ;) docstring etc.  probably unique names to both since in PyMVPA we then import everything into the suite into a flat namespace there